File: shared/src/commonMain/kotlin/dev/haroldjose/familysharedlist/domainLayer/models/FamilyListModel.kt
package dev.haroldjose.familysharedlist.domainLayer.models

import dev.haroldjose.familysharedlist.getPlatform
import kotlinx.datetime.Clock
import kotlinx.datetime.Instant
import kotlinx.datetime.LocalDateTime
import kotlinx.datetime.TimeZone
import kotlinx.datetime.toLocalDateTime


data class FamilyListModel(
    var uuid: String = "",
    var name: String,
    var isCompleted: Boolean = false,
    var isCompletedDate: LocalDateTime? = null,
    var isPrioritized: Boolean = false,
    var quantity: Int = 1,
    var price: Double = 0.0,
    var product: ProductModel? = null,
) {
    init {
        this.uuid = uuid.ifEmpty { getPlatform().generateUUID() }
    }
}

File: iosApp/iosApp/extensions/models/FamilyListModel.swift
import shared

extension FamilyListModel: Identifiable {
    
    public var id: String {
        self.uuid
    }
    
    convenience init(
        name: String,
        isCompleted: Bool = false,
        isCompletedDate: Kotlinx_datetimeLocalDateTime? = nil,
        isPrioritized: Bool = false,
        quantity: Int32 = 1,
        price: Double = 0,
        product: ProductModel? = nil
    ) {
        self.init(
            uuid: "",
            name: name,
            isCompleted: isCompleted, 
            isCompletedDate: isCompletedDate,
            isPrioritized: isPrioritized,
            quantity: quantity,
            price: price,
            product: product
        )
    }
}
